chalicelib/utils.py

```diff
@@ -44,8 +47,10 @@ def workspace_exists(self):
                        Authorization=self.auth)
         r = requests.get(self.url, headers=headers)
         if r.status_code == 200:
+            self.statusCode['status codes']['workspace exists'] = True
             return True
         else:
+            self.statusCode['status codes']['workspace exists'] = False
```

Comment:
The check for a workspace exists could fail for several reasons.
1. It doesn't exist (404)
2. Your token is invalid/not present/not authorized (401, 403)
3. Backend is having issues (5xx)
4. Possibly an illegal workspace name (not sure what those are are if they exist) -- probably some 4xx.
5.???

Which I think means you should look for a 404 as proof that the workspace does not exists. If you get anything other than a 200 or 404, then you should return the error that FC is returning.

Reply:
Thanks. I'll put those changes in today.